pilot/pkg/model/extensions.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package model
+
+import (
+	"strings"
+	"time"
+
+	envoy_config_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoy_extensions_filters_http_wasm_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/wasm/v3"
+	envoy_extensions_wasm_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/wasm/v3"
+	"github.com/gogo/protobuf/types"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/any"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/durationpb"
+
+	extensions "istio.io/api/extensions/v1alpha1"
+	"istio.io/istio/pkg/config"
+	"istio.io/istio/pkg/util/gogo"
+)
+
+const (
+	defaultRuntime = "envoy.wasm.runtime.v8"
+)
+
+type WasmPluginWrapper struct {
+	extensions.WasmPlugin
+
+	Name      string
+	Namespace string
+
+	ExtensionConfiguration *envoy_extensions_filters_http_wasm_v3.Wasm
+}
+
+func convertToWasmPluginWrapper(plugin *config.Config) *WasmPluginWrapper {
+	if wasmPlugin, ok := plugin.Spec.(*extensions.WasmPlugin); ok {
+		cfg := &any.Any{}
+		if wasmPlugin.PluginConfig != nil && len(wasmPlugin.PluginConfig.Fields) > 0 {
+			cfgJSON, _ := protojson.Marshal(proto.MessageV2(wasmPlugin.PluginConfig))
+			cfg = gogo.MessageToAny(&types.StringValue{
+				Value: string(cfgJSON),
+			})
+		}
+		var datasource *envoy_config_core_v3.AsyncDataSource
+		if strings.HasPrefix(wasmPlugin.Url, "http") {
```

Comment:
is https allowed?

Reply:
I think technically the agent supports it, that's why I used `HasPrefix` - didn't have a chance to test it yet though. For a real solution this will need to actually parse the url properly and look at the scheme, as we want to support docker:// as well (and probably even default to that if there's no scheme)